pkg/apis/stackconfigpolicy/v1alpha1/stackconfigpolicy_types.go

```diff
@@ -0,0 +1,210 @@
+	// +kubebuilder:pruning:PreserveUnknownFields
+	ComponentTemplates *commonv1.Config `json:"componentTemplates,omitempty"`
+	// ComposableIndexTemplates holds the Index Templates settings (/_index_template)
+	// +kubebuilder:pruning:PreserveUnknownFields
+	ComposableIndexTemplates *commonv1.Config `json:"composableIndexTemplates,omitempty"`
+}*/
+
+type StackConfigPolicyStatus struct {
+	// ResourcesStatuses holds the status for each resource to be configured.
+	ResourcesStatuses map[string]ResourcePolicyStatus `json:"resourcesStatuses"`
+	// Resources is the number of resources to be configured.
+	Resources int `json:"resources,omitempty"`
+	// Ready is the number of resources successfully configured.
+	Ready int `json:"ready,omitempty"`
+	// Errors is the number of resources which have an incorrect configuration
+	Error int `json:"errors,omitempty"`
+	// ReadyCount is a Human representation of the number of resources successfully configured.
+	ReadyCount string `json:"readyCount,omitempty"`
+	// Phase is the phase of the StackConfigPolicy.
+	Phase PolicyPhase `json:"phase,omitempty"`
+	// ObservedGeneration is the most recent generation observed for this StackConfigPolicy.
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+}
+
+type PolicyPhase string
+
+const (
+	UnknownPhase         PolicyPhase = "Unknown"
+	ReadyPhase           PolicyPhase = "Ready"
+	ApplyingChangesPhase PolicyPhase = "ApplyingChanges"
+	InvalidPhase         PolicyPhase = "Invalid"
+	ErrorPhase           PolicyPhase = "Error"
+	ConflictPhase        PolicyPhase = "Conflict"
+)
+
+// phaseOrder
+var phaseOrder = map[PolicyPhase]int{
+	UnknownPhase:         -1,
+	ReadyPhase:           0,
+	ApplyingChangesPhase: 1,
+	InvalidPhase:         2,
+	ErrorPhase:           3,
+	ConflictPhase:        4,
+}
+
+// ResourcePolicyStatus models the status of the policy for one resource to be configured.
+type ResourcePolicyStatus struct {
+	Phase           PolicyPhase       `json:"phase,omitempty"`
+	CurrentVersion  int64             `json:"currentVersion,omitempty"`
+	ExpectedVersion int64             `json:"expectedVersion,omitempty"`
+	Error           PolicyStatusError `json:"error,omitempty"`
+}
+
+type PolicyStatusError struct {
+	Version int64    `json:"version,omitempty"`
+	Errors  []string `json:"errors,omitempty"`
+}
+
+func NewStatus(scp StackConfigPolicy) StackConfigPolicyStatus {
+	status := StackConfigPolicyStatus{
+		ResourcesStatuses:  map[string]ResourcePolicyStatus{},
+		Phase:              ReadyPhase,
+		ObservedGeneration: scp.Generation,
+	}
+	status.setReadyCount()
+	return status
+}
+
+func (s *StackConfigPolicyStatus) setReadyCount() {
+	s.ReadyCount = fmt.Sprintf("%d/%d", s.Ready, s.Resources)
+}
+
+func (s *StackConfigPolicyStatus) AddPolicyErrorFor(resource types.NamespacedName, phase PolicyPhase, msg string) {
+	s.ResourcesStatuses[resource.String()] = ResourcePolicyStatus{
+		Phase: phase,
+		Error: PolicyStatusError{Errors: []string{msg}},
+	}
```

Comment:
What if a status already exists for this named resource?  Since `PolicyStatusError.Errors` is a slice of error messages, should it be appended to if exists with the same `Phase`?